File: kotlin/dnd-character/src/main/kotlin/DndCharacter.kt
import kotlin.math.floor
import kotlin.random.Random

class DndCharacter {

    val strength: Int = ability()
    val dexterity: Int = ability()
    val constitution: Int = ability()
    val intelligence: Int = ability()
    val wisdom: Int = ability()
    val charisma: Int = ability()
    val hitpoints: Int = 10 + modifier(constitution)

    companion object {

        fun ability(): Int =
            List(4) { Random.nextInt(1, 7) }.sorted().drop(1).sum()


        fun modifier(score: Int): Int = floor((score - 10.0) / 2).toInt()
    }

}

File: python/dnd-character/dnd_character.py
from math import floor
from random import randrange


class Character:
    def __init__(self):
        self.strength = self.ability()
        self.dexterity = self.ability()
        self.constitution = self.ability()
        self.intelligence = self.ability()
        self.wisdom = self.ability()
        self.charisma = self.ability()
        self.hitpoints = 10 + modifier(self.constitution)

    def ability(self):
        return sum(sorted([randrange(1, 7) for _ in range(4)])[1:])



def modifier(ability):
    return floor((ability - 10) / 2)
